addons/mail/static/src/core/common/attachment_model.js

```diff
@@ -32,7 +32,7 @@ export class Attachment extends FileModelMixin(Record) {
     }
 
     get isDeletable() {
-        if (this.message && !this.store.self.isInternalUser) {
+        if (this.message && this.store.self.main_user_id?.share !== false) {
```

Comment:
Why `!== false` here? To me, every positive check should be `this.store.self.main_user_id?.share` as they are often use for access check, in which case `undefined` should not allow access either.

Reply:
This is a general problem in JS with optional chaining when testing for "not false". We don't want `undefined` to be included. There is no other way to correctly write these conditions, unless splitting to condition without optional chaining to clearly indicate that should be done when there is no user, which in most cases should be to return false.

Note that `share` is the inverse of internal user, so we need a negation/checking `false` specifically.